backend/api/Controllers/InspectionFindingsController.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Api.Controllers.Models;
+using Api.Database.Models;
+using Api.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers
+{
+    [ApiController]
+    [Route("inspection-findings")]
+    public class InspectionFindingsController : ControllerBase
+    {
+        private readonly IInspectionService _inspectionService;
+        private readonly ILogger<InspectionFindingsController> _logger;
+        public InspectionFindingsController(
+            ILogger<InspectionFindingsController> logger,
+            IInspectionService inspectionService
+        )
+        {
+            _logger = logger;
+            _inspectionService = inspectionService;
+
+        }
+
+        /// <summary>
+        /// Associate a new inspection finding with the inspection corresponding to isarStepId
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        [HttpPost("add-findings")]
+        [Authorize(Roles = Role.Admin)]
+        [ProducesResponseType(typeof(InspectionFindings), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<InspectionFindings>> AddFindings([FromBody] InspectionFindingsQuery inspectionFinding)
+        {
+            _logger.LogInformation("Updating inspection findings for inspection with isarStepId '{Id}'", inspectionFinding.IsarStepId);
+            try
+            {
+                var inspection = await _inspectionService.ReadByIsarStepId(inspectionFinding.IsarStepId);
```

Comment:
This returns an "Inspection" and not an "InspectionFindings" as stated in the return type, should perhaps return the list of inspectionfindings on a isarStepId

Reply:
No, this is for fetching the inspection object based on the isarStepId and then we add inspectionfindings to this object.

Reply:
I think it should return inspection as it is required in the next line of code.

Reply:
The check on the next line ois based on the premise that there should not be a null inspection to avoid adding any Rogue inspection findings which are not associated with any inspections.

Reply:
Done now. Hence resolving. We agreed that we return the finding for the post call and inspection for the getcall.